third_party/terraform/resources/resource_container_cluster.go.erb

```diff
@@ -1145,21 +1152,15 @@ func resourceContainerClusterRead(d *schema.ResourceData, meta interface{}) erro
 		return err
 	}
 
-	cluster := &containerBeta.Cluster{}
-	err = resource.Retry(2*time.Minute, func() *resource.RetryError {
-		name := containerClusterFullName(project, location, d.Get("name").(string))
-		cluster, err = config.clientContainerBeta.Projects.Locations.Clusters.Get(name).Do()
-		if err != nil {
-			return resource.NonRetryableError(err)
-		}
-		if cluster.Status != "RUNNING" {
-			return resource.RetryableError(fmt.Errorf("Cluster %q has status %q with message %q", d.Get("name"), cluster.Status, cluster.StatusMessage))
-		}
-		return nil
-	})
+	clusterName := d.Get("name").(string)
+	name := containerClusterFullName(project, location, clusterName)
+	cluster, err := config.clientContainerBeta.Projects.Locations.Clusters.Get(name).Do()
 	if err != nil {
 		return handleNotFoundError(err, d, fmt.Sprintf("Container Cluster %q", d.Get("name").(string)))
 	}
+	if cluster.Status == "ERROR" || cluster.Status == "DEGRADED" {
```

Comment:
I think this means that if a cluster gets into one of these states that it will be impossible to fix them with terraform. Since plan/apply/destroy will all error on refresh.

Reply:
Yep! This is sad and also effectively what we did before at https://github.com/GoogleCloudPlatform/magic-modules/pull/2021/files#diff-29c7fa35b303cd012accf212d201bd5dL1155 if the state the cluster was in was terminal.

Reply:
cool, just ruminating out loud.